xarray/core/options.py

```diff
@@ -41,6 +69,21 @@ def __init__(self, **kwargs):
             raise ValueError('argument names %r are not in the set of valid '
                              'options %r' % (invalid_options, set(OPTIONS)))
         self.old = OPTIONS.copy()
+
+        if 'io_engines' in kwargs:
+            if isinstance(kwargs['io_engines'], basestring):
+                kwargs['io_engines'] = [kwargs['io_engines']]
+            new_engines = list()
+            for e in kwargs['io_engines']:
+                if e in _AVAILABLE_IO_ENGINES:
+                    new_engines.append(e)
+                else:
+                    raise ValueError('I/O engine %s not installed' % e)
```

Comment:
since you only raise an error for invalid engines, I think this logic could be simplified:
```
new_engines  = list(kwargs['io_engines'])
for engine in new_engines:
    if engine not in _AVAILABLE_IO_ENGINES:
        raise ValueError(...)
```